server/datastore/mysql/apple_mdm.go

```diff
@@ -1527,6 +1530,78 @@ func (ds *Datastore) UpdateOrDeleteHostMDMAppleProfile(ctx context.Context, prof
 	return err
 }
 
+func (ds *Datastore) SetVerifiedHostMacOSProfiles(ctx context.Context, host *fleet.Host, installedProfiles []*fleet.HostMacOSProfile) error {
+	installedProfsByIdentifier := make(map[string]*fleet.HostMacOSProfile, len(installedProfiles))
+	for _, p := range installedProfiles {
+		installedProfsByIdentifier[p.Identifier] = p
+	}
+
+	var teamID uint
+	if host.TeamID != nil {
+		teamID = *host.TeamID
+	}
+	var expectedProfs []*fleet.MDMAppleConfigProfile
+	if err := sqlx.SelectContext(ctx, ds.reader, &expectedProfs, `
+SELECT
+	name,
+	identifier,
+	updated_at
+FROM
+	mdm_apple_configuration_profiles
+WHERE
+	team_id = ?`, teamID); err != nil {
+		return ctxerr.Wrap(ctx, err, "listing expected profiles for update host macOS profiles")
+	}
+
+	verifiedProfs := make([]*fleet.MDMAppleConfigProfile, 0, len(expectedProfs))
+	for _, ep := range expectedProfs {
+		ip, ok := installedProfsByIdentifier[ep.Identifier]
+		if !ok {
+			// TODO: expected profile is not installed on host, skip it for now
+			continue
+		}
+		if ep.UpdatedAt.After(ip.InstallDate) {
+			// TODO: host has an older version of expected profile installed, skip it for now
+			continue
+		}
+		if ep.Name != ip.DisplayName {
+			// TODO: host has a different name for expected profile, skip it for now
+			continue
```

Comment:
not sure about this use case. My hunch is that if the `ep.UpdatedAt.After(ip.InstallDate)` above is working, we shouldn't be checking this at all.

did you see anything strange with the display name?

Reply:
Hmm I don't know, if the osquery query did have a hash of the content we would use it, and if the display name is different than the expected name then the hash would be different, so to me this is a good additional indication that it's not the right profile that was deployed on the host? Unless we're not totally sure the DisplayName in the osquery is indeed the `PayloadName` of the profile.